broker-core/src/main/java/io/zeebe/broker/subscription/message/state/MessageStateController.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Zeebe Broker Core
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.broker.subscription.message.state;
+
+import io.zeebe.broker.util.KeyStateController;
+import java.io.File;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
+import org.agrona.DirectBuffer;
+import org.agrona.ExpandableArrayBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.RocksDB;
+
+public class MessageStateController extends KeyStateController {
+
+  private static final byte[] EXISTENCE = new byte[] {1};
+
+  private static final int TIME_OFFSET = 0;
```

Comment:
The thing is that we need this for MessageSubscriptions as well.

Reply:
Ah, then it makes sense. I just wanted to avoid that we have change the data type in the Message class and forget to update the offset here. It's fine as is then.